misc/python/materialize/checks/mzcompose_actions.py

```diff
@@ -51,6 +51,12 @@ def execute(self, e: Executor) -> None:
             {"enable_specialized_arrangements": "false"} if self.tag else None
         )
 
+        catalog_store = (
+            "shadow"
+            if not self.tag or self.tag >= MzVersion.parse("0.78.0-dev")
+            else "stash"
+        )
```

Comment:
So my worry here is a test like the following:

1. Start Materialize in version 0.77.0 -> use the "stash" catalog.
2. Stop Materialize.
3. Start Materialize in version 0.78.0 -> use the "shadow" catalog.
4. Panic because the "shadow" catalog will look for data in persist that doesn't exist.

The issue is that the test needs to use a consistent value for `catalog_store` every time is starts and already initialized Materialize.

Reply:
Yeah, I think this is why the random upgrade tests are panicking in the most recent nightlies.

Reply:
I think I can fix that, but it's a bit more involved.

Reply:
I hope it's good now, another run: https://buildkite.com/materialize/nightlies/builds/5133

Reply:
Oh awesome, thanks so much! I think I can even revert my changes to `misc/python/materialize/checks/scenarios_upgrade.py`

Reply:
True, they shouldn't be required. I'll leave that up to you now, pushed into your branch enough for today :D

Reply:
I pushed the change and re-triggered Nightly. It looks like it's working.